likes/listeners.py

```diff
@@ -0,0 +1,32 @@
+def incr_likes_count(sender, instance, created, **kwargs):
+    from tweets.models import Tweet
+    from django.db.models import F
+
+    if not created:
+        return
+
+    model_class = instance.content_type.model_class()
+    if model_class != Tweet:
+        # TODO HOMEWORK 给 Comment 使用类似的方法进行 likes_count 的统计
+        return
+
+    # 不可以使用 tweet.likes_count += 1; tweet.save() 的方式
+    # 因此这个操作不是原子操作，必须使用 update 语句才是原子操作
+    # SQL: UPDATE likes_count = likes_count + 1 FROM tweets_table WHERE id = instance.object_id, 有 row lock
+
+    tweet = instance.content_object
+    Tweet.objects.filter(id=tweet.id).update(likes_count=F('likes_count') + 1)
```

Comment:
f means use likes_count in the query

Reply:
we only change the likes count in db, not touching the cache yet. It's because if every time we need to invalidate the cache when the like count changed, it's too easy to invalidate the cache. we need to find the other way